src/cpp/session/SessionRUtil.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * SessionRUtil.cpp
+ *
+ * Copyright (C) 2009-2015 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include <session/SessionRUtil.hpp>
+
+#include <core/Error.hpp>
+#include <core/Log.hpp>
+#include <core/Exec.hpp>
+#include <core/Macros.hpp>
+
+#include <core/FileSerializer.hpp>
+
+#include <r/RExec.hpp>
+
+#include <session/SessionModuleContext.hpp>
+
+#include <boost/regex.hpp>
+
+#include "modules/shiny/SessionShiny.hpp"
+
+namespace rstudio {
+
+using namespace core;
+
+namespace session {
+namespace r_utils {
+
+namespace {
+
+boost::regex& reYaml()
+{
+   static boost::regex instance("^[\\s\\n]*---\\s*(.*?)---\\s*(?:$|\\n)");
+   return instance;
+}
+
+} // anonymous namespace
+
+bool hasYamlHeader(const FilePath& filePath)
```

Comment:
Do we plan on keeping `rmarkdown` embedded (so that we can always be sure that this method is available)?